tmk_core/protocol/chibios/usb_main.c

```diff
@@ -381,14 +381,17 @@ static uint16_t get_hword(uint8_t *p) {
  * Other Device    Required    Optional    Optional    Optional    Optional    Optional
  */
 
-#if defined(SHARED_EP_ENABLE) && !defined(KEYBOARD_SHARED_EP)
 static uint8_t set_report_buf[2] __attribute__((aligned(2)));
 static void    set_led_transfer_cb(USBDriver *usbp) {
-    if ((set_report_buf[0] == REPORT_ID_KEYBOARD) || (set_report_buf[0] == REPORT_ID_NKRO)) {
-        keyboard_led_stats = set_report_buf[1];
+    if (usbp->setup[6] == 2) { /* LSB(wLength) */
```

Comment:
My understanding is this is trying to do the same as <https://github.com/qmk/qmk_firmware/blob/6b17067b1571bd19023e175980b8d74765ce7cfe/tmk_core/protocol/lufa/lufa.c#L476>, But is the indexing here always going to be safe?

Reply:
Yep. This `setup` structure is part of the USB spec: https://www.beyondlogic.org/usbnutshell/usb6.shtml
I don't know why ChibiOS didn't create a struct for it, though.
http://chibios.sourceforge.net/docs3/hal/struct_u_s_b_driver.html#a221310f11fd99174b053218141f26fcc